nx-X11/programs/Xserver/hw/nxagent/NXmiexpose.c

```diff
@@ -710,6 +710,12 @@ int what;
     ChangeGCVal gcval[7];
     ChangeGCVal newValues [COUNT_BITS];
 
+    /* don't let these undefined before using them */
+    int t;
+    for (t=0;t<COUNT_BITS;t++) {
+	newValues[t].ptr = NULL;
+	newValues[t].val = 0;
+    }
```

Comment:
I think that instead of this loop, you could just replace the line
```C
    ChangeGCVal newValues [COUNT_BITS];
```

With
```C
    ChangeGCVal newValues [COUNT_BITS] = {{ 0 }};
```

Making use of the universal 0 initializer (two braces, because we're initializing an array [first brace] of structs [second brace] with a scalar as the first element [no more braces required].)